pkg/otlp/model/source/source_provider.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"fmt"
+)
+
+// Kind of source
+type Kind string
+
+const (
+	// InvalidKind is an invalid kind. It is the zero value of Kind.
+	InvalidKind Kind = ""
```

Comment:
I'd use the Go-style (see https://pkg.go.dev/net/http constants for StatusCode* as one example) way of declaring constants representative of a type, and that is prefixing them with the type name (or some form of it), e.g.:
* `KindInvalid`
* `KindHostname`
* `KindAWSECSFargate`